ldt/src/ldt_based_pcs.rs

```diff
@@ -128,32 +128,38 @@ where
     #[instrument(name = "prove batch opening", skip_all)]
     fn open_multi_batches(
         &self,
-        prover_data_and_points: &[(&Self::ProverData, &[EF])],
+        prover_data_and_points: &[(&Self::ProverData, &[Vec<EF>])],
         challenger: &mut Challenger,
     ) -> (OpenedValues<EF>, Self::Proof) {
         let coset_shift: Val =
             <Self as UnivariatePcsWithLde<Val, EF, In, Challenger>>::coset_shift(self);
 
         // Use Barycentric interpolation to evaluate each matrix at a given point.
-        let eval_at_point = |matrices: &[M::Mat<'_>], point| {
-            matrices
+        let eval_at_points = |matrix: M::Mat<'_>, points: &[EF]| {
+            points
                 .iter()
-                .map(|mat| {
-                    let (low_coset, _) = mat.split_rows(mat.height() >> self.ldt.log_blowup());
+                .map(|&point| {
+                    let (low_coset, _) =
+                        matrix.split_rows(matrix.height() >> self.ldt.log_blowup());
                     interpolate_coset(&BitReversedMatrixView::new(low_coset), coset_shift, point)
                 })
-                .collect::<OpenedValuesForPoint<EF>>()
+                .collect::<OpenedValuesForMatrix<EF>>()
         };
 
         let all_opened_values = info_span!("compute opened values with Lagrange interpolation")
             .in_scope(|| {
                 prover_data_and_points
                     .iter()
-                    .map(|(data, points)| {
+                    .map(|(data, points_per_matrix)| {
                         let matrices = self.mmcs.get_matrices(data);
-                        points
+
+                        matrices
                             .iter()
-                            .map(|&point| eval_at_point(&matrices, point))
+                            .enumerate()
```

Comment:
nit: could we zip with `points` to avoid needing indices?

Reply:
I think the other two uses of `enumerate()` could also potentially use `zip()`

Reply:
Definitely ! Thanks for the suggestions